cpp/src/Manager.cpp

```diff
@@ -1290,7 +1290,7 @@ uint8 Manager::GetNodeDeviceType
 {
 	if( Driver* driver = GetDriver( _homeId ) )
 	{
-		return driver->GetNodeDeviceType( _nodeId );
+		return (uint8)driver->GetNodeDeviceType( _nodeId );
```

Comment:
this function expects a uint8 as return, so then this function header should be changed